libvips/foreign/heifsave.c

```diff
@@ -311,6 +311,14 @@ vips_foreign_save_heif_write_page(VipsForeignSaveHeif *heif, int page)
 	}
 #endif /*HAVE_HEIF_ENCODING_OPTIONS_OUTPUT_NCLX_PROFILE*/
 
+#ifdef HAVE_HEIF_ENCODING_OPTIONS_IMAGE_ORIENTATION
+	/* EXIF orientation is informational in the HEIF specification.
+	 * Orientation is defined using irot and imir transformations.
+	 */
+	options->image_orientation = vips_image_get_orientation(save->ready);
+	vips_autorot_remove_angle(save->ready);
```

Comment:
This is likely redundant since `save->ready` is a private (final) copy of `save->in` and only used when the image is written to an output file, buffer or memory.
```suggestion
```

Reply:
The idea behind removing the Orientation property, if any, from EXIF when writing an image is to reduce the chance of double-rotation (or similar confusion) when reading the image back again.

Reply:
Ah, I just noticed that `vips_autorot_remove_angle()` also removes the `exif-ifd0-Orientation` field, if any.

Indeed, that makes sense and would prevent accidental double rotations, although EXIF orientation tags are purely informational according to the HEIF standard. :+1: